benchmark/C/Savina/src/concurrency/Dictionary.lf

```diff
@@ -218,24 +154,24 @@ reactor Worker(instance:int(0), num_requests:int(10), write_percentage:int(10)){
     =}
 }
 
-main reactor(num_workers:int(20),num_requests:int(8000),write_percentance:int(10)){
-
-	/* [[[cog
- 	  cog.outl(f'dictionary = new Dictionary(num_workers={num_workers});')
+/* [[[cog
+  	  cog.outl(f'main reactor(numWorkers:int({numWorkers}),numMessagesPerWorker:int({numMessagesPerWorker}),writePercentage:int({writePercentage}))')
 	]]] */
-    dictionary = new Dictionary(num_workers=num_workers);
+    main reactor(numWorkers:int(20),numMessagesPerWorker:int(8000),writePercentage:int(10))
 	// [[[end]]]
-	workers = new[num_workers] Worker(num_requests=num_requests,write_percentage=write_percentance);
+{
+
+    dictionary = new DictionaryImpl(num_workers=numWorkers);
+	workers = new[numWorkers] Worker(num_requests=numMessagesPerWorker,write_percentage=writePercentage);
 	
 	workers.read->dictionary.read;
 	workers.write->dictionary.write;
 	dictionary.result->workers.result;
 	
 	reaction(startup)->dictionary.start, workers.start{=
```

Comment:
Similar to other benchmarks, we don't really need the start ports for C and both the dictionary and the workers could react directly to startup.